src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -324,30 +326,33 @@ private static IResourceBuilder<ProjectResource> WithProjectDefaults(this IResou
         }
         else
         {
-            // If we aren't a web project (looking at both launch profile and Kestrel config) we don't automatically add bindings.
-            if (launchProfile?.ApplicationUrl == null && !kestrelEndpointsByScheme.Any())
+            // Set HTTP_PORTS/HTTPS_PORTS in publish mode, to override the default port set in the base image. Note that:
+            // - We don't set them if we have Kestrel endpoints configured, as Kestrel will get everything from its config.
+            // - We only do that for endpoint set explicitely (.WithHttpEndpoint), not for the ones coming from launch profile.
```

Comment:
This is what we discussed with @davidfowl and @DamianEdwards, but let's make sure it's the right behavior.